liveRecorder/Tasks/ConcatinationTask.py

```diff
@@ -124,6 +133,100 @@ def get_flavor_id(self, url_postfix, single_flavor):
+                            remote_dc_type_str = 'PRIMARY'
+                    except ValueError as e:
+                        self.logger.fatal("invalid content in dc file: {}".format(str(e)))
+                        raise e
+                    except Exception as e:
+                        self.logger.fatal("failed to read or parse dc file: {}".format(str(e)))
+                        raise e
+
+                    self.logger.debug('successfully read dc file. Entry server node id [{}], server type [{}]'.format(
+                        entry_server_node_id, local_dc_type_str))
+
+                    response_list, response_header = self.backend_client.get_server_entry_nodes_list(self.entry_id)
+                    self.logger.info('List length:{}'.format(len(response_list.objects)))
+                    ''' if len(response_list.objects) == 1:
+                        self.logger.debug(
+                            'only one dc returned from call to server nodes list. Recording job will be processed')
+                        should_process = True'''
+
+                    if len(response_list.objects) == 0:
+                        self.logger.debug('recording job was processed by LIVE {} DC'.format(remote_dc_type_str))
+                        should_process = False
+                    elif int(self.duration) >= 1000 * max_duration_sec:
+                        self.logger.debug(
+                            "Duration is [{}] msec, processing vod since reached max allowed".format(self.duration))
+                    else:
+                        local_duration = int(self.duration)
+                        max_duration = local_duration
+                        id_of_processing_dc = entry_server_node_id
+                        for entry_server_node in response_list.objects:
+                            if entry_server_node.id != entry_server_node_id:
+                                for recorded_entry_info in entry_server_node.recordingInfo:
+                                    if entry_server_node.recordingInfo and recorded_entry_info.recordedEntryId == self.recorded_id:
+                                        recording_status = KalturaEntryServerNodeRecordingStatus(entry_server_node.recordingInfo.recordingStatus)
+                                        if recording_status == KalturaEntryServerNodeRecordingStatus.ON_GOING:
+                                            max_duration = recorded_entry_info.duration
+                                            id_of_processing_dc = entry_server_node.id
+                                            break
+                                        duration_diff_sec = (local_duration - recorded_entry_info.duration) / 1000
+                                        duration_is_equal = abs(duration_diff_sec) <= self.duration_diff_tolerance_sec
+                                        if not duration_is_equal and recorded_entry_info.duration > local_duration:
```

Comment:
where is the code that favors the primary instead of backup if +-same duration?

Reply:
check now and you'll find it easily, in should_process_as_redundant_dc(), line 240